dev-docs/metrics.md

```diff
@@ -179,3 +179,98 @@ Strong references to instruments will be discarded when changes to the aggregate
 On the fast path the mutex is locked for the period that it takes to upgrade the weak reference. This is a fast operation, and should not block the thread for any meaningful period of time.
 
 If there is shown to be contention in future profiling we can revisit.
+
+## Adding new metrics
+There are different types of metrics.
+
+* Static - Used by us to monitor feature usage.
+* Dynamic - Defined by the user to monitor the health and performance of their system.
+
+> New features should add BOTH static and dynamic metrics.
+
+> Why are static metrics no longer recommended for users to use directly?
+> 
+> They can, but usually it'll be only a starting point for them. We can't predict the things that users will want to monitor, and if we tried we would blow up the cardinality of our metrics resulting in high costs for our users via their APMs.
```

Comment:
that is not clear to me. What do we mean by "users using static metrics directly?" Is it when they would add that in their custom plugin? (which would not increase cardinality for all users) Or asking us to add a new metric to the router?

Reply:
[df05add](https://github.com/apollographql/router/pull/5116/commits/df05add809a8dade4a720b338a2cbc1ef082ea1e)